### ISSUE TITLE
Levenberg Marquardt as Keras loss function

### ISSUE THREAD

**Issue by swilson314** [tensorflow/graphics#156]
I've been using LM for many years and we currently use it for our in-house deep learning code. We are in the process of switching over to tf/keras and I'd like to continue using LM, which we often find produces better models than Adam when the datasets are small enough for LM. There are a handful of stackoverflow posts by other people who want to do this, but I haven't seen a resolution anywhere. This surprises me given that LM is designed to minimize MSE. Do you have any insight to why this has proven difficult and why LM is not already part of keras?

**Comment by fulkast** [comment]
Out is curiosity, what's the typical size of a model that you mentioned being small enough?

**Comment by swilson314** [comment]
I've got a 96 core server with a ton of ram. Largest dataset I'm running is 1.5M cases with 2400 float inputs. Network is 5 layers and lower level layers use convolution that decreases number of variables by factor of 20.

**Comment by sofienbouaziz** [comment]
This is because computing a Jacobian using backpropagation is inefficient. For that one might want to use forward propagation but it is not implemented in Tensorflow.

**Comment by sofienbouaziz** [comment]
The TensorFlow Graphics implementation relies on calling multiple time backpropagation to build the Jacobian. I guess this could be ported as a keras optimizer but it would be useful only for small problems as it is not very efficient.

**Comment by rola93** [comment]
Hey!


I'm having exact the same issue, converting an old matlab script to keras, need to run levenberg_marquardt optimier.

I'd appreciate any clues on how to implement it. I found [levenberg-marquardt](https://www.tensorflow.org/graphics/api_docs/python/tfg/math/optimizer/levenberg_marquardt) implementation for this optimizer in [tensorflow_graphics](https://www.tensorflow.org/graphics/api_docs/python) but it isnt compatible with keras optimizers. 

Any help is welcome

**Comment by fabiodimarco** [comment]
I also had the same issue. I have found that in some cases LM produces better results than first-order methods.
I decided to implement a tf/keras version of LM, bringing some additional features.
Here is the link to the repository: [tf-levenberg-marquardt](https://github.com/fabiodimarco/tf-levenberg-marquardt).
Hope it helps.
